nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/data/explorer/PutAzureDataExplorer.java

```diff
@@ -0,0 +1,399 @@
+                ingestionMappingKind = IngestionProperties.DataFormat.TSVE.getIngestionMappingKind();
+                break;
+            case TXT:
+                ingestionProperties.setDataFormat(IngestionProperties.DataFormat.TXT);
+                ingestionMappingKind = IngestionProperties.DataFormat.TXT.getIngestionMappingKind();
+                break;
+        }
+
+        if (StringUtils.isNotEmpty(context.getProperty(MAPPING_NAME).getValue()) && ingestionMappingKind != null) {
+            ingestionProperties.setIngestionMapping(context.getProperty(MAPPING_NAME).getValue(), ingestionMappingKind);
+        }
+
+        ingestionProperties.setReportLevel(IngestionProperties.IngestionReportLevel.FAILURES_AND_SUCCESSES);
+        ingestionProperties.setReportMethod(IngestionProperties.IngestionReportMethod.TABLE);
+        ingestionProperties.setFlushImmediately(false);
+        ingestionProperties.setIgnoreFirstRecord(StringUtils.equalsIgnoreCase(context.getProperty(IGNORE_FIRST_RECORD).getValue(), IGNORE_FIRST_RECORD_YES.getValue()));
+
+        boolean isError = false;
+
+        // ingestion into ADX
+        try (final InputStream inputStream = session.read(flowFile)) {
+            StringBuilder ingestLogString = new StringBuilder().append("Ingesting with: ")
+                    .append("dataFormat - ").append(ingestionProperties.getDataFormat()).append("|")
+                    .append("ingestionMapping - ").append(ingestionProperties.getIngestionMapping().getIngestionMappingReference()).append("|")
+                    .append("reportLevel - ").append(ingestionProperties.getReportLevel()).append("|")
+                    .append("reportMethod - ").append(ingestionProperties.getReportMethod()).append("|")
+                    .append("databaseName - ").append(ingestionProperties.getDatabaseName()).append("|")
+                    .append("tableName - ").append(ingestionProperties.getTableName()).append("|")
+                    .append("flushImmediately - ").append(ingestionProperties.getFlushImmediately());
+            getLogger().info(ingestLogString.toString());
+
+            KustoIngestionResult result = service.ingestData(new KustoIngestionRequest(isStreamingEnabled, pollOnIngestionStatus, inputStream, ingestionProperties));
+            getLogger().info("Poll on Ingestion Status {} and Operation status: {} ", pollOnIngestionStatus, result.toString());
+
+            if (result == KustoIngestionResult.SUCCEEDED) {
+                getLogger().info("Operation status Succeeded - {}", result.toString());
+            } else if (result == KustoIngestionResult.FAILED) {
+                getLogger().error("Operation status Error - {}", result.toString());
+                isError = true;
+            } else if (result == KustoIngestionResult.PARTIALLY_SUCCEEDED) {
+                getLogger().error("Operation status Partially succeeded - {}", result.toString());
```

Comment:
Considering partially succeeded status as success might result in silent data loss when the user ignores warning logs for a huge set of data, Hence added a new "partial success" relationship where the partially succeeded data can be routed and appropriate action can be taken by the user. What are your thoughts on this?

Reply:
Thanks for flagging this scenario @tanmaya-panda1. What would the user be expected to do in this scenario? The problem seems to be knowing what was ingested and what was not ingested. One option might be to introduce a configurable property for `Partial Success Routing`, with values of `success` or `failure`. This would let the user decide on a high-level strategy. In addition, the status should be noted in a FlowFile attribute, which would support routing decisions that could include retrying.